app/models/conditions_response/membership_fee_overdue_alert.rb

```diff
@@ -0,0 +1,21 @@
+# This emails a member _after_ their membership_fee payment is due.
+# It warns them that their membership_fee payment is overdue.
+#
+class MembershipFeeOverdueAlert < UserEmailAlert
+
+
+  def self.send_alert_this_day?(timing, config, user)
+
+    return false unless user.has_approved_shf_application?
```

Comment:
To prevent unnecessary processing:
```
return false if user.membership_current? ||  !user.has_approved_shf_application?
```

Reply:
2 issues with doing it that way:
1. check for the application status first because if the user has a _rejected_ application (or in any state other than accepted), there's no reason to check whether or not the membership is current.  (checking if it is current = gets into payments and a boatload of SQL queries)

2. We currently have members that are "legacy" members: they _are_ members but they are past due or haven't paid their membership fee (according to data in our system).   How to handle these?  Are they current or not?  (and what is the date that == day 1 for counting them as overdue?   This is the discussion we need to have)

Reply:
My understanding of (2) above is wrong.

Reply:
yes,  let's just reverse the order of your suggested gating methods.